src/useLocalStorageState.ts

```diff
@@ -68,6 +78,13 @@ function useLocalStorage<T>(
         parsed: undefined,
     })
 
+    const [error, setError] = useState<null | Error>(null)
+
+    // Nice little React short cut - we can ignore rules of hooks if we are throwing errors.
+    if(error){
```

Comment:
So this makes sense to me, but I'm wondering if it's a bit unconventional.... do people expect hooks to throw errors?